R/data.table.R

```diff
@@ -1912,8 +1911,8 @@ as.matrix.data.table <- function(x, rownames=NULL, rownames.value=NULL, ...) {
       # TODO in future as warned in NEWS for 1.11.6:
       #   warning("length(rownames)>1 is deprecated. Please use rownames.value= instead")
       if (length(rownames)!=nrow(x))
-        stop(sprintf("length(rownames)==%d but nrow(DT)==%d. The rownames argument specifies a single column name or number. Consider rownames.value= instead.",
-                     length(rownames), nrow(x)))
+        stop("length(rownames)==", length(rownames), " but nrow(DT)==", nrow(x),
```

Comment:
`stop(sprintf`/`stop(paste` construction to be avoided

Reply:
I like `stop(sprintf(`, it is easier to follow than switching context between an error string and R expressions in it, but I am ok with this change.

Reply:
I generally agree it's more readable, just taking heed from [WRE](https://cran.r-project.org/doc/manuals/r-release/R-exts.html)

> in R error messages do not construct a message with `paste` (such messages will not be translated) but via multiple arguments to `stop` or `warning`, or via `gettextf`.